src/pages/LoginPage/useLoginForm.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo, useState } from 'react';
+
+export const useLoginForm = () => {
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const handleEmailChange = (value: string) => {
+    setEmail(value);
+    setEmailError('');
+  };
+
+  const handlePasswordChange = (value: string) => {
+    setPassword(value);
+    setPasswordError('');
+  };
+
+  const validateEmail = () => {
+    if (!email.trim()) {
+      setEmailError('ID를 입력해주세요.');
+      return false;
+    }
+    if (!/^[\w.-]+@[a-zA-Z\d.-]+\.[a-zA-Z]{2,}$/.test(email)) {
+      setEmailError('ID는 이메일 형식으로 입력해주세요.');
+      return false;
+    }
+    return true;
+  };
```

Comment:
```suggestion
  const error = useMemo(() => {
    if (!email.trim()) {
      return 'ID를 입력해주세요.';
    }
    if (!/^[\w.-]+@[a-zA-Z\d.-]+\.[a-zA-Z]{2,}$/.test(email)) {
      return 'ID는 이메일 형식으로 입력해주세요.';
    }
    return '';
  }, ['email']);
```

이렇게 파생 상태로 관리해도 되지 않을까요!?